common/lib/transport/connectionmanager.js

```diff
@@ -408,6 +410,14 @@ var ConnectionManager = (function() {
 		if(connectionKey && this.connectionKey != connectionKey)  {
 			this.setConnection(connectionId, connectionKey, connectionSerial);
 		}
+		if(clientId) {
+			if(this.realtime.clientId && this.realtime.clientId != clientId) {
+				/* Should never happen in normal circumstances as realtime should
+				 * recognise mismatch and return an error */
+				Logger.logAction(Logger.LOG_ERROR, 'ConnectionManager.activateTransport()', 'Unexpected mismatch between expected and received clientId');
```

Comment:
Sorry, I think I confused/misled you with my last comment on this. I agree this should never happen but if it does :) it's a fatal error I think so we should fail the entire connection.